### DISCUSSION TITLE
VS 22 or other competition

### DISCUSSION THREAD

**Discussion by TinyTimothy2** [TabularEditor/TabularEditor3#1281] [🙋♀️TE3 Q&A]
I don't have V22 yet.  Is TE3 still better than Visual Studio 22?  Is there anything in Power BI or Fabric that would render TE3 superfluous?
My concern on buying 1 year is that a MS tool will supplant TE3.
Thoughts?

**Comment by otykier** [comment]
As a Tabular Editor employee, I am of course biased, but my take is this:

VS 2022 (or any other new version of VS) is not going to improve the existing experience of VS. Remember that creating semantic models in VS is done through the Analysis Services Projects extension, and to my knowledge, no improvements have been made to that extension, so it doesn’t matter which version of Visual Studio you’re using.

Power BI Desktop has seen some improvements recently, such as the DAX Query View and Copilot. However, for large, complex models, nothing beats the “offline” experience of Tabular Editor, with undo/redo, copy/paste, lightning fast batch changes, etc. You will still see the infamous “Working on it…”-spinner in both Power BI Desktop and the Power BI/Fabric web modeling, every time you make a change to the model. This is not going to change anytime soon as that would require a fundamental rewrite of the entire architecture behind these tools.

But if you believe that these tools will eventually surpass TE3, by all means stick to a monthly subscription. It’s easy to cancel anytime. Or alternatively, use the free TE2 as a supplement to Microsoft’s tools. No one is forcing you to buy a TE3 license. There is no lock-in or anything like that.

**Reply by mlonsk** [reply]
With the same caveat as @otykier and fully agreeing with his answer - Tabular Editor 3 also provides features that you are unlikely to ever get in Power BI nor VS 2022.

To name a few:
- Scripting and macros
- Offline metadata analysis with syntax/semantic checking
- VertiPaq Analyzer
- Async Data Refresh
- DAX Debugger
- Multi monitor support (Drag your windows wherever you want)
- Customizable shortcuts 
- ...

I think Daniel's point about no lock-in is particular salient as with the new .pbip you can even save existing .bim or TMDL models into a Power BI desktop file if you want.

**Comment by edhans** [comment]
I think it is worth mentioning that I _believe_ part of the reason TE exists in the first place is the VS experience is so horrible, and it is truly horrible. I have created one model using VS 2019 with the Analysis Services Project Extension and it was the worst developer experience I've ever had. I'll take TE/2 over the VS experience any day of the week. TE/3 makes it an absolute no brainer.

To compete with TE, MS wouldn't need to improve the experience. They would need to start from scratch with an entirely different UI and workflow.
